volmdlr/shells.py

```diff
@@ -118,11 +118,23 @@ def __init__(self, faces: List[volmdlr.faces.Face3D],
         self._vertices_graph = None
         self._vertices_points = None
         self._shell_octree_decomposition = None
+        self.index = 0
 
         volmdlr.core.CompositePrimitive3D.__init__(self,
                                                    primitives=faces, color=color, alpha=alpha,
                                                    reference_path=reference_path, name=name)
 
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.index >= len(self.faces):
+            self.index = 0
+            raise StopIteration
+        else:
+            self.index += 1
+            return self.faces[self.index-1]
```

Comment:
Because in the previous if it raises an error, you don't need to use an else statement here. You can delete the indentation of the block after it.